wandb/sdk/data_types.py

```diff
@@ -1627,8 +1627,8 @@ def _initialize_from_data(self, data: "ImageDataType", mode: str = None,) -> Non
         else:
             if hasattr(data, "numpy"):  # TF data eager tensors
                 data = data.numpy()
-            if data.ndim > 2:
-                data = data.squeeze()  # get rid of trivial dimensions as a convenience
+            if data.ndim > 2 and data.shape[2] == 1:
```

Comment:
it looks like we are trying to squeeze out the extra dimensions at the end? So if there are more than 3 dimensions, this might not work properly. Maybe the following would catch those cases too?
```
while data.ndim > 2 and data.shape[-1] == 1: data = data.squeeze(axis=len(data.shape)-1)
```

Reply:
Hmm, maybe we should only do this if `data.ndim == 3`.  @bycn any thoughts?

Reply:
To me, this is more of an invalid input, I would prefer an informative error i.e. `assert data.ndim < 4, "Invalid img shape x"`. As far as I know, wandb is only supporting logging 1 image at a time. It seems that the most common case somebody would have 4 dims would be [N, H, W, C] in which case they would be logging the wrong thing anyways.